rotkehlchen/chain/ethereum/modules/convex/convex_cache.py

```diff
@@ -0,0 +1,140 @@
+
+
+class ConvexPoolData(NamedTuple):
+    """mapping of convex pool rewards address to name"""
+    pool_address: ChecksumEvmAddress
+    pool_name: str
+
+
+def get_existing_pools(cursor: DBCursor) -> set[ChecksumEvmAddress]:
+    return {
+        string_to_evm_address(address) for address in globaldb_get_general_cache_values(
+            cursor=cursor,
+            key_parts=(CacheType.CONVEX_POOL_ADDRESS,),
+        )
+    }
+
+
+def save_convex_data_to_cache(
+        write_cursor: DBCursor,
+        database: 'DBHandler',  # pylint: disable=unused-argument
+        new_data: list[ConvexPoolData],
+) -> None:
+    """Stores data about convex pools and their names in the global db cache tables."""
+    for pool in new_data:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.CONVEX_POOL_ADDRESS,),
+            values=[pool.pool_address],
+        )
+        globaldb_set_unique_cache_value(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.CONVEX_POOL_NAME, pool.pool_address),
+            value=pool.pool_name,
+        )
+
+
+def read_convex_data_from_cache() -> tuple[dict[ChecksumEvmAddress, str], set[ChecksumEvmAddress]]:
+    """Reads convex pools and names from global db cache tables."""
+    pools: dict[ChecksumEvmAddress, str] = {}
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        for pool_address in get_existing_pools(cursor):
+            if (pool_name := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(CacheType.CONVEX_POOL_NAME, pool_address),
+            )) is None:
+                continue
+            pools[pool_address] = pool_name
+
+    return pools, set()  # no gauges
+
+
+def query_convex_data_from_chain(
+        ethereum: 'EthereumInquirer',
+        existing_pools: set[ChecksumEvmAddress],
+) -> dict[ChecksumEvmAddress, str]:
+    """
+    Query Booster contract and fetch all reward pools. It creates a mapping for convex pools
+    to the symbol of the underlying lp token.
+
+    May raise:
+    - RemoteError if failed to query chain
+    """
+    booster_contract = ethereum.contracts.contract(BOOSTER)
+    pools_count = booster_contract.call(node_inquirer=ethereum, method_name='poolLength')
+    calls_to_booster = [(
+        booster_contract.address,
+        booster_contract.encode('poolInfo', [x]),
+    ) for x in range(pools_count)]
+    booster_result = ethereum.multicall(calls=calls_to_booster)
+    convex_rewards_addrs: list[ChecksumEvmAddress] = []
+    convex_lp_tokens_addrs = []
+    for single_booster_result in booster_result:
+        lp_token_addr = hex_or_bytes_to_address(single_booster_result[0:32])
+        crv_rewards = hex_or_bytes_to_address(single_booster_result[96:128])
+        if crv_rewards not in existing_pools:
+            convex_rewards_addrs.append(crv_rewards)
+            convex_lp_tokens_addrs.append(lp_token_addr)
```

Comment:
is it the `crv_rewards` or the `lp_token_addr` what exists in `existing_pools`?

```suggestion
        crv_rewards = hex_or_bytes_to_address(single_booster_result[96:128])
        if crv_rewards not in existing_pools:
            lp_token_addr = hex_or_bytes_to_address(single_booster_result[0:32])
            convex_rewards_addrs.append(crv_rewards)
            convex_lp_tokens_addrs.append(lp_token_addr)
```